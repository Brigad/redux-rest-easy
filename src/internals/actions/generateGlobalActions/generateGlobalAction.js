const generateGlobalAction = actionType => () => ({
  type: actionType,
});

export default generateGlobalAction;
