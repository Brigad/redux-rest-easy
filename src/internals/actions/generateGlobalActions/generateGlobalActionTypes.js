import generateActionType from '../../utils/generateActionType';

const generateGlobalActionTypes = () => ({
  RESET_ALL: generateActionType('@global', '@reset', 'RESET_ALL'),
});

export default generateGlobalActionTypes;
