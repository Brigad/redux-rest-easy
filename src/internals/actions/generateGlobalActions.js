import generateGlobalAction from './generateGlobalActions/generateGlobalAction';
import generateGlobalActionTypes from './generateGlobalActions/generateGlobalActionTypes';

const generateGlobalActions = () => {
  const { RESET_ALL } = generateGlobalActionTypes();

  return {
    resetAll: generateGlobalAction(RESET_ALL),
  };
};

export default generateGlobalActions;
