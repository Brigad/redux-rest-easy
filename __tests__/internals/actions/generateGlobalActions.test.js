import generateGlobalActions from '../../../src/internals/actions/generateGlobalActions';

describe('generateGlobalActions', () => {
  test('only path', () => {
    expect(generateGlobalActions()).toMatchSnapshot();
  });
});
