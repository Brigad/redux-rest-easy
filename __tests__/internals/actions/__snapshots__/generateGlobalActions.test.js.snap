// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateGlobalActions only path 1`] = `
Object {
  "resetAll": [Function],
}
`;
